pkg/segment/query/segquery_test.go

```diff
@@ -408,19 +409,22 @@ func Test_segQueryFilter(t *testing.T) {
 	numEntriesForBuffer := 10
 	fileCount := 5
 	instrumentation.InitMetrics()
-	config.InitializeTestingConfig(t.TempDir())
+	dir := t.TempDir()
+	config.InitializeTestingConfig(dir)
+	segBaseDir, _, err := writer.GetMockSegBaseDirAndKeyForTest(dir, "metadatafilter")
+	assert.NoError(t, err)
 	limit.InitMemoryLimiter()
-	err := InitQueryNode(getMyIds, serverutils.ExtractKibanaRequests)
+	err = InitQueryNode(getMyIds, serverutils.ExtractKibanaRequests)
 	if err != nil {
 		t.Fatalf("Failed to initialize query node: %v", err)
 	}
-	metadata.InitMockColumnarMetadataStore("data/", fileCount, numBuffers, numEntriesForBuffer)
+	metadata.InitMockColumnarMetadataStore(segBaseDir, fileCount, numBuffers, numEntriesForBuffer)
 
 	bloomMetadataFilter(t, numBuffers, numEntriesForBuffer, fileCount)
 	rangeMetadataFilter(t, numBuffers, numEntriesForBuffer, fileCount)
 	// pqsSegQuery(t, numBuffers, numEntriesForBuffer, fileCount)
 	// add more simple, complex, and nested metadata checking
 	time.Sleep(1 * time.Second) // sleep to give some time for background pqs threads to write out dirs
-	err = os.RemoveAll("data/")
+	err = os.RemoveAll(dir)
```

Comment:
nit: you don't really need to remove this dir since it's a `TempDir()` on line 412. The testing framework should take care of deleting the dir. But it's fine to keep this